docs/components/table-view.md

```diff
@@ -0,0 +1,1016 @@
+/>
+
+```html
+<KTableView
+  :cell-attrs="({ headerKey, row }) => { return { 'data-testid': `column-${headerKey}-row-${row.id}` } }"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### maxHeight
+
+Pass a number (px) you want to limit table height to. If the height table exceeds the specified number, it will be scrollable. Table header is a `position: sticky;` element and will always be visible.
+
+<KTableView
+  max-height="300"
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  max-height="300"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+## States
+
+### Empty
+
+<KTableView 
+  :data="[]"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView 
+  :data="[]"
+  :headers="headers"
+/>
+```
+
+Set the following props to handle empty state:
+
+- `emptyStateTitle` - Title text for empty state
+- `emptyStateMessage` - Message for empty state
+- `emptyStateIconVariant` - Icon variant for empty state (see [KEmptyState component props](/components/empty-state#iconvariant))
+- `emptyStateActionMessage` - Button text for empty state action
+- `emptyStateActionRoute` - Route for empty state action
+- `emptyStateButtonAppearance` - Appearance of empty state action button. See [KButton `appearance` prop](/components/button#appearance) for details
+
+:::tip
+Should you want to display an icon inside of action button, you can use `empty-state-action-icon` slot.
+:::
+
+When empty state action button is clicked, KTableView emits the `empty-state-action-click` event.
+
+<KTableView 
+  :data="[]"
+  :headers="basicHeaders()"
+  empty-state-title="No Workspaces exist"
+  empty-state-message="Adding a new Workspace will populate this catalog."
+  empty-state-icon-variant="kong"
+  empty-state-action-message="Create a Workspace"
+  empty-state-action-route="/"
+  empty-state-button-appearance="secondary"
+>
+  <template #empty-state-action-icon>
+    <AddIcon />
+  </template>
+</KTableView>
+
+```html
+<KTableView 
+  :data="[]"
+  :headers="headers"
+  empty-state-title="No Workspaces exist"
+  empty-state-message="Adding a new Workspace will populate this table."
+  empty-state-icon-variant="kong"
+  empty-state-action-message="Create a Workspace"
+  empty-state-action-route="/"
+  empty-state-button-appearance="secondary"
+>
+  <template #empty-state-action-icon>
+    <AddIcon />
+  </template>
+</KTableView>
+```
+
+### Error
+
+Set the `error` prop to `true` to enable the error state.
+
+<KTableView
+  error
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  error
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+Set the following properties to customize error state:
+
+- `errorStateTitle` - Title text for error state
+- `errorStateMessage` - Message for error state
+- `errorStateActionMessage` - Button text for error state action
+- `errorStateActionRoute` - Route for error state action
```

Comment:
issue/question: is `errorStateActionRoute` a new prop for this component? What if it's an external link? the interface defined is just `Object | string` this seems wrong?

Reply:
It's not new - migrated over from KTable

Reply:
It's a very loose interface 🤔

Reply:
Agreed, pushed an improvement